chalk-solve/src/infer/instantiate.rs

```diff
@@ -109,12 +110,16 @@ pub(crate) trait IntoBindersAndValue {
     fn into_binders_and_value(self) -> (Self::Binders, Self::Value);
 }
 
-impl<'a, T> IntoBindersAndValue for &'a Binders<T> {
+impl<'a, I, T> IntoBindersAndValue for (&'a Binders<T>, &'a I)
+where
+    I: Interner,
+    T: HasInterner<Interner = I>,
+{
     type Binders = std::iter::Cloned<std::slice::Iter<'a, ParameterKind<()>>>;
     type Value = &'a T;
 
     fn into_binders_and_value(self) -> (Self::Binders, Self::Value) {
```

Comment:
hmm, I think we should modify the trait here to be:

```rust
pub trait IntoBindersAndValue<I: Interner> {
    fn into_binders_and_value(self, &I)
```

instead of making the impl be over a tuple like this

Reply:
Done, adding a lifetime parameter.